internal/services/storagemover/storage_mover_agent_resource.go

```diff
@@ -0,0 +1,227 @@
+package storagemover
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/storagemover/2023-03-01/agents"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/storagemover/2023-03-01/storagemovers"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	computevalidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type StorageMoverAgentResourceModel struct {
+	Name                string `tfschema:"name"`
+	StorageMoverId      string `tfschema:"storage_mover_id"`
+	ArcVirtualMachineId string `tfschema:"arc_virtual_machine_id"`
+	ArcVmUuid           string `tfschema:"arc_vm_uuid"`
+	Description         string `tfschema:"description"`
+}
+
+type StorageMoverAgentResource struct{}
+
+var _ sdk.ResourceWithUpdate = StorageMoverAgentResource{}
+
+func (r StorageMoverAgentResource) ResourceType() string {
+	return "azurerm_storage_mover_agent"
+}
+
+func (r StorageMoverAgentResource) ModelObject() interface{} {
+	return &StorageMoverAgentResourceModel{}
+}
+
+func (r StorageMoverAgentResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return agents.ValidateAgentID
+}
+
+func (r StorageMoverAgentResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"arc_virtual_machine_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: computevalidate.HybridMachineID,
+		},
+
+		"arc_vm_uuid": {
```

Comment:
For consistency could we spell this out instead of abbreviating it
```suggestion
		"arc_virtual_machine_uuid": {
```

Reply:
Fixed.